src/main/java/core/basesyntax/services/StorageOperation.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.services;
+
+import java.time.LocalDate;
+
+public interface StorageOperation {
+    public boolean updateTransactionTable(String filePath, String type,
```

Comment:
let's refactor the logic of your communication with storage please

let's use the strategy pattern:
you have the StorageOperation interface, let's have only one `apply(Transaction transaction)` method here

create three classes implementing this interface: SupplyOperation, BuyOperation and ReturnOperation: their apply() method will call the necessary action related to the **storage**

create a StorageService with methods to _add product, remove product and get all available products_ (you will use this method when you will need to write the available products into an output file)
inside you will have your exact **storage** as a static field - try to implement this as a `Map<String, Map<LocalDate, Integer>>`

in your SupplyOperation, BuyOperation and ReturnOperation let's have a StorageService field which methods you will call inside `apply()`

after you have parsed the lines of your csv file and made a Transaction instance out of each line - check the `operation` field of your Transaction instance and based on its value choose the necessary StorageOperation implementation and call `apply()`

this will call a suitable StorageService method and perform the necessary action